sdk/packages/flight-sql-client/src/lib.rs

```diff
@@ -29,14 +29,14 @@ use crate::flight_client::{execute_flight, setup_client, ClientOptions};
 
 fn init_logging() {
     // Set up a subscriber that logs to stdout
-    let subscriber = tracing_subscriber::FmtSubscriber::builder()
+    if let Err(result) = tracing_subscriber::FmtSubscriber::builder()
         .with_env_filter(
             EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
         )
-        .finish();
-
-    // Set the global default subscriber
-    tracing::subscriber::set_global_default(subscriber).expect("Failed to set tracing subscriber");
```

Comment:
this line seems to go away entirely?  Is it just not needed?  What was it doing before?

Reply:
Yeah, the [`try_init()`](https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/fmt/mod.rs.html#1156) does the same thing in a way that's safer to use in libraries.
```
/// Install a global tracing subscriber that listens for events and
/// filters based on the value of the [`RUST_LOG` environment variable],
/// if one is not already set.
```

Reply:
`tracing::subscriber::set_global_default()` is appropriate to use in applications, not libraries (like our SDK).

Reply:
https://docs.rs/tracing/latest/tracing/index.html#in-executables
```
Warning: In general, libraries should not call set_global_default()! Doing so will cause conflicts when executables that depend on the library try to set the default later.
```